draft-gpew-priv-ppm.md

```diff
@@ -672,247 +677,375 @@ independently.
+Input share preparation consists of running the preparation-state initialization
+algorithm for the VDAF associated with the task and computes the first state transition.
+This produces three possible values:
 
-[OPEN ISSUE: consider updating the `checksum` algorithm to not permit
-collisions]
+1. An error, in which case the input report share is marked as invalid.
+1. An output share, in which case the aggregator stores the output share for future collection
+   as described in {{collect-flow}}.
+1. An initial VDAF state and preparation message, denoted `(prep_state, prep_msg)`.
 
-First, the leader computes a checksum over the set of output shares included in
-the batch window. The checksum is computed by taking the SHA256 hash of each
-nonce from the client reports included in the aggregation, then combining the
-hash values with a bitwise-XOR operation.
+Each aggregator runs this procedure for a given input share with corresponding nonce
+as follows:
 
-Then, for each aggregator endpoint `[aggregator]` in the parameters associated
-with `CollectReq.task_id` (see {{pa-collect}}) except its own, the leader sends
-a POST request to `[aggregator]/aggregate_share` with the following message:
+~~~
+prep_state = VDAF.prep_init(vdaf_verify_param, agg_param, nonce, input_share)
+out = VDAF.prep_next(prep_state, None)
+~~~
+
+`vdaf_verify_param` is the public VDAF parameter, and `agg_param` is the opaque aggregation
+parameter. If either step fails, the aggregator marks the report as invalid with error
+`vdaf-prep-error`.
+
+Otherwise, the value `out` is interpreted as follows. If this is the last round of the VDAF,
+then `out` is the aggregator's output share. Otherwise, `out` is the pair `(prep_state, prep_msg)`.
+
+### Aggregate Continuation {#agg-continue-flow}
+
+In the continuation phase, the leader drives the VDAF preparation of each share in the
+candidate report set until the underlying VDAF moves into a terminal state, yielding
+an output share for all aggregators or an error. This phase may involve multiple rounds
+of interaction depending on the underlying VDAF parameters. Each round trip is initiated
+by the leader.
+
+#### Leader Continuation
+
+The leader begins each round of continuation for a report share based on its locally computed
+prepare message and the previous PrepareStep from the helper. If PrepareStep is of type "failed",
+then the leader marks the report as failed and removes it from the candidate report set and does not
+process it further. If the type is "finished", then the leader aborts with "unrecognizedMessage".
+[[OPEN ISSUE: This behavior is not specified.]] If the type is "continued", then the leader proceeds as
+follows.
+
+Let `leader_outbound` denote the leader's prepare message and `helper_outbound` denote the
+helper's. The leader computes the next state transition as follows:
+
+~~~
+inbound = VDAF.prep_shares_to_prep(agg_param, [leader_outbound, helper_outbound])
+out = VDAF.prep_next(prep_state, inbound)
```

Comment:
I think we had Process at some point in this PR, but that was confusing since it didn't align with the VDAF terminology. I suggest we punt this to the VDAF draft and then just use whatever term makes sense there.

Reply:
https://github.com/cjpatton/vdaf/issues/48